src/mpi/coll/igather/igather_tsp_tree_algos.h

```diff
@@ -138,7 +138,7 @@ int MPIR_TSP_Igather_sched_intra_tree(const void *sendbuf, int sendcount,
     } else if (num_children > 0 && lrank != 0) {        /* intermediate ranks in the tree */
         tmp_buf = MPIR_TSP_sched_malloc(recv_size * sendtype_extent, sched);
     } else {    /* leaf ranks in the tree */
-        tmp_buf = sendbuf;
+        tmp_buf = (void *) sendbuf;
```

Comment:
does https://github.com/pmodels/mpich/pull/3202/commits/31f735f9166ee69529185920f58377b6a624feb1#diff-26ac474ae73830c50d705bbd0dd5c1c0R157 not have the same issue?

Reply:
Sure. Maybe I missed the warning or compiler didn't warn about it.
PR updated with just this change.